src/hotspot/share/nmt/mallocHeader.inline.hpp

```diff
@@ -121,33 +124,36 @@ inline const MallocHeader* MallocHeader::resolve_checked(const void* memblock) {
 inline bool MallocHeader::looks_valid() const {
   // Note: we define these restrictions loose enough to also catch moderately corrupted blocks.
   // E.g. we don't check footer canary.
-  return ( (_canary == _header_canary_live_mark NOT_LP64(&& _alt_canary == _header_alt_canary_live_mark)) ||
-           (_canary == _header_canary_dead_mark NOT_LP64(&& _alt_canary == _header_alt_canary_dead_mark)) ) &&
-           _size > 0 && _size < max_reasonable_malloc_size;
+  return ( (canary() == _header_canary_live_mark NOT_LP64(&& alt_canary() == _header_alt_canary_live_mark)) ||
+           (canary() == _header_canary_dead_mark NOT_LP64(&& alt_canary() == _header_alt_canary_dead_mark)) ) &&
+           size() > 0 && size() < max_reasonable_malloc_size;
 }
 
 inline bool MallocHeader::check_block_integrity(char* msg, size_t msglen, address* p_corruption) const {
```

Comment:
This function should not even be called for ASAN, so it should not need any ASAN code.

NMT comes with its own canary system, which is intended for use when there is no ASAN and we have no other way to check block integrity.

But with ASAN, when touching either header or footer, we should get an ASAN crash right away. That is better than what NMT can do.